### PR TITLE
2.4.x `_utter_responses` should handle `template` and `response` 

### PR DESCRIPTION
**Pull request by yennycheung**
**Proposed changes**:
- https://github.com/RasaHQ/rasa/pull/8226 but pointed towards 2.4.x
- I cherry-picked it to a new branch because it was a rebase hell.
- Fixes https://github.com/RasaHQ/rasa/issues/8223

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by wochinge — APPROVED** [review]
💯

**Comment by yennycheung** [comment]
@wochinge Got a problem with the test, I'll look into it.

File: rasa/core/actions/action.py [hunk 1]
```diff
@@ -625,7 +625,17 @@ async def _utter_responses(
         """Use the responses generated by the action endpoint and utter them."""
         bot_messages = []
         for response in responses:
+            generated_response = None
             generated_response = response.pop("response", None)
+            if not generated_response:
+                generated_response = response.pop("template", None)
+                rasa.shared.utils.io.raise_deprecation_warning(
```
Comment (wochinge) [inline rasa/core/actions/action.py:632]: should we only raise this warning if `generated_response` from the `template` key `is not None`? If they for some reason send neither of these keys then this warning could be confusing.

**Review by wochinge — APPROVED** [review]

File: rasa/core/actions/action.py (hunk 1 above)
Comment (yennycheung) [reply rasa/core/actions/action.py:632]: Great point, fixed

**Review by wochinge — APPROVED** [review]

**Pull request merged**